components/github-api-client/src/client.rs

```diff
@@ -27,10 +28,52 @@ use serde_json;
 
 use config::GitHubCfg;
 use error::{HubError, HubResult};
+use metrics::Counter;
 use types::*;
 
 const USER_AGENT: &'static str = "Habitat-Builder";
 
+pub type TokenString = String;
+pub type InstallationId = u32;
+
+/// Bundle up a Github token with the Github App installation ID used
+/// to create it.
+///
+/// Consumers will treat this as an opaque type; its main utility is
+/// in carrying the installation ID around so we can generate metrics
+/// on a per-installation basis.
+pub struct AppToken {
+    inner_token: TokenString,
+
+    // Leave this here in anticipation of using it for tagged metrics
+    #[allow(dead_code)]
+    installation_id: InstallationId,
+}
+
+impl AppToken {
+    // Not public, because you should only get these from
+    // `GitHubClient::app_installation_token`
+    fn new(inner_token: TokenString, installation_id: InstallationId) -> Self {
+        AppToken {
+            inner_token,
+            installation_id,
+        }
+    }
+
+    // Only providing this for builder-worker's benefit... it
+    // currently needs a token for cloning via libgit2; once that's
+    // gone, just delete this function.
+    /// Retrieve the actual token content for use in HTTP calls.
+    pub fn inner_token(&self) -> &str {
+        self.inner_token.as_ref()
+    }
+}
+
+// Temporary type... will go away soon once we transition away from
+// personal Github tokens. Useful for expressing expectations through
+// explicitly typing, though.
+pub type UserToken = TokenString;
```

Comment:
This'll work great, however if it was a [newtype](https://rustbyexample.com/generics/new_types.html)-style struct, simply deleting this type later would show every place `UserToken` was used. I'm not 100% sure either way, but I believe that by type aliasing `UserToken` to `String` (ultimately), that the compiler swaps the alias name for the underlying type (as well as exposes all methods of the underlying type via the alias). Maybe someone will check me on this--it's one reason I use the "newtype" pattern so much.

Reply:
Ah, yeah... that's probably a good tweak to make. In the one callsite we have, I did use `(token as UserToken)`, so we'll still get a compiler error when it gets removed, but that behavior is up to the programmer to remember 😄 

Good call.